### PR TITLE
Material door updates: lock and damage

### PR DESCRIPTION
**Pull request by Alberyk**
-ported locks and keys from baystation
-fixed the simple door damage interaction, now you can properly destroy it and etc

### CODE REVIEW

File: code/modules/lock/key.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+/obj/item/weapon/key/soap
+	name = "soap key"
+	desc = "a fragile key made using a bar of soap."
+	var/uses = 0
+
+/obj/item/weapon/key/soap/get_data(var/mob/user)
+	uses--
+	if(uses == 1)
+		to_chat(user, "<span class='warning'>\The [src] is going to break soon!</span>")
```
Comment (BurgerLUA) [inline code/modules/lock/key.dm]: I think that the notification should be removed. Not even emags show their charge left.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -1513,6 +1513,24 @@
 	var/obj/effect/golemrune/Z = new /obj/effect/golemrune
 	Z.forceMove(get_turf(holder.my_atom))
 
+/datum/chemical_reaction/soap_key
+	name = "Soap Key"
+	id = "skey"
+	result = null
+	required_reagents = list("frostoil" = 2, "cleaner" = 5)
```
Comment (BurgerLUA) [inline code/modules/reagents/Chemistry-Recipes.dm]: Perhaps frost oil is a bit overkill here considering how rare the chemical actually is.

Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Recipes.dm]: I suggest using triglycerides as one of the ingredients. Most soap is made with oil/fat.

One of the other ingredients is lye, which is from ashes. If you're willing to make a new chem to go along with this that other servers have, that would be going above and beyond but I think an alternative could possibly be salt or something.

Water is also common in soap as well.

I suggest: 1 part triglycerides, 1 part water, 1 part space cleaner. These are easy to obtain chemicals.

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -79,6 +79,7 @@
 
 	var/cleaving = FALSE
 	var/reach = 1 // Length of tiles it can reach, 1 is adjacent.
+	var/lock_picking_level = 0 //used to determine whether something can pick a lock, and how well.
```
Comment (MarinaGryphon) [inline code/game/objects/items.dm]: I'm not so sure about adding another variable to every `/obj/item/`.

**Pull request merged**